test/AreaEstimator/utest_area_estimator.cpp

```diff
@@ -0,0 +1,152 @@
+#include <gtest/gtest.h>
+#include <iostream>
+
+#include "area_estimator_test_utils.h"
+
+Rectangle cell = read_cell(1, -1, -1, 1);
```

Comment:
Test naming is good and I suggest extra improvement: use a pattern for a test name (the 2nd parameter in the TEST macro).
You can use the following: <direction>_<how a beam passes a cell>_<whether the cell is occupied>_<extra info>?.
For example, Left_Right_ParallelToSide_ThroughCenter -> LR-SideAligned_ThroughCenter_Empty (Left, Right, Diagonal and other common for the "test suite" words can be replaced by their first letters)